File: 45주차/폰켄몬.js
function solution(nums) {
  let dict = {};
  for (var v of nums) {
    if (v in dict) dict[v] += 1;
    else dict[v] = 0;
  }

  return Math.min(Object.keys(dict).length, nums.length / 2);
}
// 방법2
function sol(nums) {
  var answer = 0;
  let a = new Set(nums);
  console.log(a);
  console.log(nums.length / 2);
  answer = Math.min(a.size, nums.length / 2);

  return answer;
}
// 플그머 해시 문제

File: 47주차/폰켄몬.cpp
#include <vector>
#include <map>
#include <cmath>
using namespace std;

int solution(vector<int> nums)
{
    map<int,int> p;
    
    int answer = 0;
    
    for (int i = 0; i< nums.size(); i++) {
        p[nums[i]] = 1;
    }

    answer = min(p.size(),nums.size()/2);
    return answer;
}
//플그머 c++ 고득점 kit 문제